pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -560,7 +595,7 @@ func (nc *NodeController) MonitorNodeStatus() error {
 				nc.deletePods(node.Name)
 			}
 			if lastReadyCondition.Status == api.ConditionUnknown &&
-				nc.now().After(lastReadyCondition.LastProbeTime.Add(nc.podEvictionTimeout-gracePeriod)) {
+				nc.now().After(nc.nodeStatusMap[node.Name].timestamp.Add(nc.podEvictionTimeout-gracePeriod)) {
```

Comment:
This will work only if we assume that skew is roughly constant. I have absolutely no idea how this currently works, and if it's possible that for some reason it can rapidly change. 

OTOH if we assume that the skew don't change a lot it may be simpler to just account for that by adding Kubelet 'push' period to the difference between first observed timestamp and local NodeController time, and then using this amount as an 'expected' lag. This way we'll add few seconds to latency, but I'm not sure if that it's critical for NodeController. In the same time we'll gain some simplicity (simple per-node 'expected lag' period instead of this constantly updated timestamp.